### PR TITLE
[Refactor] Refactor models to support HF hub

### PR DESCRIPTION
**Pull request by philipphager**
Our model architecture had to mirror the [FlaxBertPreTrainedModel](https://github.com/huggingface/transformers/blob/1773afcec338c2b1a741a86b7431ad10be4518c7/src/transformers/models/bert/modeling_flax_bert.py#L760) to support model storing using HF hub. Essentially, the PreTrainedModel class is a HF wrapper that contains all the logic for initializing and serializing a model, and it wraps inside the actual Flax module that we want to customize. Unfortunately, we cannot just use the `FlaxBertPreTrainedModel` wrapper and replace the inner Flax module, as we need custom model inputs. Thus, I added a custom `FlaxPreTrainedCrossEncoder` wrapper class, that mirrors the HF implementation, but uses our batch input.

All models are essentially now just Flax modules that are placed inside a `FlaxPreTrainedCrossEncoder`  wrapper and thus can be pushed / downloaded from HF using e.g.:

```Python
model = CrossEncoder.from_pretrained("...")

# Model then transforms to:
batch = next(iter(click_loader))
model(batch)

# Or only for rel prediction:
batch = next(iter(rels_loader))
model.predict_relevance(batch)

```

I also added minor changes to remove the base BERT model, slight updates to the loss tracking, and I dropped models that we did not end up using (RetroMAE, CoCondenser).

### Pushing to HF
We can push the trained model to HF by changing the config to, e.g.:
```
hf_hub_push: True
hf_hub_user: philipphager
hf_hub_model: uva-bert-cross-encoder
hf_hub_token: my-auth-token
```

Still open

- [x] Run models and compare loss
- [x] Store all models on HF
- [x] Update evaluation (maybe to directly use the models from HF)
- [x] Update data loading to use the same methods between train and eval
- [x] Fix parameter upload

### CODE REVIEW

**Comment by RomDeffayet** [comment]
Thanks :star_struck:
> Run models and compare loss
 Store all models on HF
 Update evaluation (maybe to directly use the models from HF)

I'm taking care of that

> Not sure if we should keep the batch object as input or if we should follow HF models and have separate input params like input_ids, attention_mask, etc...

I think it makes sense to keep the batch object since there is so much metadata in this dataset.
